embeddings/task/lightning_task/text_classification.py

```diff
@@ -105,9 +105,11 @@ def test_step(self, *args: Any, **kwargs: Any) -> Optional[STEP_OUTPUT]:
     def predict(
         self, dataloader: DataLoader[HuggingFaceDataset]
     ) -> Dict[str, nptyping.NDArray[Any]]:
-        predictions = torch.argmax(
-            torch.cat([self.forward(**batch).logits for batch in dataloader]), dim=1
-        ).numpy()
+        assert self.trainer is not None
```

Comment:
how about encapsulating this postprocessing in predict_step method?
https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.predict_step

Reply:
Done